packages/block-library/src/navigation-menu-item/edit.js

```diff
@@ -49,41 +45,84 @@ function NavigationMenuItemEdit( {
 	isParentOfSelectedBlock,
 	setAttributes,
 	insertMenuItemBlock,
+	fetchSearchSuggestions,
 } ) {
+	const { label, opensInNewTab, title, url } = attributes;
+	const link = title ? { title, url } : null;
 	const [ isLinkOpen, setIsLinkOpen ] = useState( false );
-	const [ isEditingLink, setIsEditingLink ] = useState( false );
-	const [ urlInput, setUrlInput ] = useState( null );
+	const [ wasClosedByLinkControl, setWasClosedByLinkControl ] = useState( false );
+
+	/**
+	 * It's a kind of hack to handle closing the LinkControl popover
+	 * clicking on the ToolbarButton link.
+	 */
+	useEffect( () => {
+		if ( ! isSelected ) {
+			setIsLinkOpen( false );
+			setWasClosedByLinkControl( false );
+		}
+		return () => {
+			setIsLinkOpen( false );
+			setWasClosedByLinkControl( false );
+		};
+	}, [ isSelected ] );
+
+	// Set the menu item when it's new.
+	// - Open LinkControl popover.
+	useEffect( () => {
+		if ( ! label && isSelected ) setIsLinkOpen( true );
+	}, [] );
 
-	const inputValue = urlInput !== null ? urlInput : url;
+	/**
+	 * `onKeyDown` LinkControl handler.
+	 * It takes over to stop the event propagation to make the
+	 * navigation work, avoiding undesired behaviors.
+	 * For instance, it will block to move between menu items
```

Comment:
I guess other use cases of LinkControl will face similar issues. Can we address this problem at the LinkControl level?

Reply:
well, it's a topic that has been quite discussed in the development of the LinkControl. At that moment we decided to delegate event handling to the component which consumes it.

However, it's a valid suggestion and we could consider delegating this handling to the LinkControl itself. We will need to analyze most scenarios as possible in order to avoid blocking some desired behaviors. For instance, how it should act in a paragraph context?